File: java/tour-of-restate/src/main/java/dev/restate/tour/auxiliary/EmailClient.java
/*
 * Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Tour of Restate Java,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/tour-of-restate
 */

package dev.restate.tour.auxiliary;

public class EmailClient {
    public static EmailClient get() {
        return new EmailClient();
    }

    public boolean notifyUserOfPaymentSuccess(String userId) {
        System.out.println("Notifying user " + userId + " of payment success");
        // send the email
        return true;
    }

    public boolean notifyUserOfPaymentFailure(String userId) {
        System.out.println("Notifying user " + userId + " of payment failure");
        // send the email
        return true;
    }
}

File: tutorials/tour-of-restate-rust/src/auxiliary/email_client.rs
// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
//
// This file is part of the Restate examples,
// which is released under the MIT license.
//
// You can find a copy of the license in the file LICENSE
// in the root directory of this repository or package or at
// https://github.com/restatedev/examples/
#![allow(unused)]

use log::info;
use restate_sdk::errors::HandlerError;

pub struct EmailClient;

impl EmailClient {
    pub async fn notify_user_of_payment_success(
        user_id: &str,
    ) -> Result<bool, HandlerError> {
        info!("Notifying user {} of payment success", user_id);
        // send the email
        Ok(true)
    }

    pub async fn notify_user_of_payment_failure(
        user_id: &str,
    ) -> Result<bool, HandlerError> {
        info!("Notifying user {} of payment failure", user_id);
        // send the email
        Ok(false)
    }
}
